frame/support/procedural/src/pallet/expand/error.rs

```diff
@@ -128,5 +191,12 @@ pub fn expand_error(def: &mut Def) -> proc_macro2::TokenStream {
 				}
 			}
 		}
+
+		impl<#type_impl_gen> #frame_support::traits::CompactPalletError
+			for #error_ident<#type_use_gen>
+			#config_where_clause
+		{
+			fn check_compactness() -> bool { #compactness_check }
+		}
```

Comment:
instead of reimplementing the derive macro here, why not directly call the macro by adding `#[derive(frame_support::CompactPalletError)` to the type Error declared by user.
We also need to derive Encode/Decode now.